### DISCUSSION TITLE
[vue warn] plugin must be a function or object with install function

### DISCUSSION THREAD

**Discussion by eshell** [quasarframework/quasar#12263] [General - Components / Directives / etc] [answered]
Hi, I'm trying to create my own "store" to use mostly for learning purposes. I'm quite new and have been looking everywhere I can think of to see what I may be doing wrong. Anyway...

I created a store, used a boot file to inject it into the global space for use in my app, and everything seems to be working fine except for that nagging vue warning. I don't know if maybe i'm missing a spot to hook up my store or what.

here's my store.js file
```
import { reactive, readonly } from 'vue'

export default function store () {
    const app = {
        name: 'Test App',
        version: '0.1'
    }

    const user = reactive({
        id: 0,
        name: '',
        loggedIn: false,
        lastPage: 'home'
    })

    const toggleLogin = () => {
        user.loggedIn = !user.loggedIn
    }

    const createUser = () => {
        if(user.id == 0) {
            user.id = 1
            user.name = 'test'
            user.loggedIn = true
        } else {
            user.id = 0
            user.name = ''
            user.loggedIn = false
        }
    }

    const setLastPage = (page) => {
        user.lastPage = page
    }

    return {
        app,
        user: readonly(user),
        toggleLogin,
        createUser,
        setLastPage
    }
}
```

Here's my boot/store.js file
```
import { boot } from 'quasar/wrappers'
import store from 'src/store'

const createStore = store()

export default boot(({ app }) => {
  app.config.globalProperties.$store = createStore
})
```

I'm successfully able to use functions and read the store from within my .vue files, I simply get that warning is what I'm struggling to understand.

**Comment by rstoenescu** [comment, marked as answer]
If you use your own custom store (awesome) then just don't use the reserved /src/store folder for it. Put it anywhere else and remove the /src/store folder. That is reserved exclusively for Vuex.

**Reply by eshell** [reply]
Ah thank you. It was driving me nuts how a function wasn't being recognized as a function. Never would have guessed it was a reserved folder for vuex if I never installed vuex.

Thanks again
